### PR TITLE
feat: add Wallet Descriptor Support for Transaction Indexing

### PR DESCRIPTION
**Pull request by qustavo**
This introduces support for wallet descriptors.
Descriptors provide a compact and semi-standardized method for describing how scripts and addresses within a wallet are generated[1].

Chainhooks users that want to track addresses derived from an extended pubkey or a multisig-wallet for example, can now rely on this feature instead of defining one predicate per address.

For example if we wanted to track the first 3 addressed generated by the following descriptor:

```
wpkh(tprv8ZgxMBicQKsPePxn6j3TjvB2MBzQkuhGgc6oRh2WZancZQgxktcnjZJ44XdsRiw3jNkbVTK9JW6KFHvnRKgAMtSyuBevMJprSkZ4PTfmTgV/84'/1'/0'/0/*)
```

which reads: describe a P2WPKH output with the specified extended public key, and produces these BIP84 addresses:

```
bcrt1qzy2rdyvu8c57qd8exyyp0mw7dk5drsu9ewzdsu
bcrt1qsfsjnagr29m8h3a3vdand2s85cg4cefkcwk2fy
bcrt1qauewfytqe5mtr0xwp786r6fl39kmum2lr65kmj
```

we could use the following predicate:

```json
...
  "networks": {
    "regtest": {
      "if_this": {
        "scope": "outputs",
        "descriptor": {
          "expression": "wpkh(tprv8ZgxMBicQKsPePxn6j3TjvB2MBzQkuhGgc6oRh2WZancZQgxktcnjZJ44XdsRiw3jNkbVTK9JW6KFHvnRKgAMtSyuBevMJprSkZ4PTfmTgV/84'/1'/0'/0/*)",
          "range": [0,3]
        }
      },
      "then_that": {
        "file_append": {
          "path": "txns.txt"
        }
      }
    }
  }
...

```

1: https://bitcoindevkit.org/descriptors/

### CODE REVIEW

**Comment by lgalabru** [comment]
This is really cool, thank you @qustavo 🚀
Do you think we could have some tests for this PR?
Happy to set-up with you and @MicaiahReid, and onboard you with our testing framework if you think it can help!

**Comment by qustavo** [comment]
> This is really cool, thank you @qustavo 🚀 Do you think we could have some tests for this PR? Happy to set-up with you and @MicaiahReid, and onboard you with our testing framework if you think it can help!

Thanks for taking the time to review, and definitively yes, I'd like to get onboard with how you test this.

**Comment by lgalabru** [comment]
@qustavo sounds great, could you shoot me a message - ludovic at hiro so? 
I'll follow up with an invite!

File: components/chainhook-sdk/Cargo.toml [hunk 1]
```diff
@@ -42,6 +42,8 @@ dashmap = "5.4.0"
 fxhash = "0.2.1"
 lazy_static = "1.4.0"
 regex = "1.9.3"
+miniscript = "10.0.0"
+bitcoin = "0.30.1"
```
Comment (MicaiahReid) [inline components/chainhook-sdk/Cargo.toml:46]: @qustavo we may not need this dep, can you confirm if we can remove?

**Comment by MicaiahReid** [comment]
Could you add a test case to the `it_handles_bitcoin_if_this_predicates` function in `components/chainhook-cli/src/service/tests/mod.rs`, as well as some documentation on this? What you have in the PR description could work great for our docs.

File: components/chainhook-sdk/Cargo.toml (hunk 1 above)
Comment (qustavo) [reply components/chainhook-sdk/Cargo.toml]: fixed

**Comment by qustavo** [comment]
> Could you add a test case to the `it_handles_bitcoin_if_this_predicates` function in `components/chainhook-cli/src/service/tests/mod.rs`, as well as some documentation on this? What you have in the PR description could work great for our docs.

added

**Review by lgalabru — APPROVED** [review]
Thanks @qustavo!

**Pull request merged**